app/src/test/java/com/fstyle/structure_android/data/source/UserRepositoryTest.java

```diff
@@ -0,0 +1,104 @@
+package com.fstyle.structure_android.data.source;
+
+import com.fstyle.structure_android.data.model.User;
+import com.fstyle.structure_android.data.model.UsersList;
+import com.fstyle.structure_android.data.source.remote.UserRemoteDataSource;
+import com.fstyle.structure_android.data.source.remote.api.service.NameApi;
+import java.util.ArrayList;
+import java.util.List;
+import okhttp3.MediaType;
+import okhttp3.ResponseBody;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import retrofit2.Response;
+import retrofit2.adapter.rxjava.HttpException;
+import rx.Observable;
+import rx.observers.TestSubscriber;
+
+/**
+ * Created by Sun on 3/11/2017.
+ */
+public class UserRepositoryTest {
+
+    private static final String USER_LOGIN_1 = "abc";
+    private static final String USER_LOGIN_2 = "def";
+
+    @Mock
+    NameApi mNameApi;
+
+    private UserRepository mUserRepository;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        mUserRepository = new UserRepository(null, new UserRemoteDataSource(mNameApi));
+    }
+
+    @After
+    public void tearDown() throws Exception {
+
+    }
+
+    @Test
+    public void searchUsers_200ResponseCode_InvokesCorrectApiCalls() {
+        // Given
+        Mockito.when(mNameApi.searchGithubUsers(ArgumentMatchers.anyString()))
+                .thenReturn(Observable.just(githubUserList()));
+
+        // When
+        TestSubscriber<UsersList> subscriber = new TestSubscriber<>();
+        mUserRepository.getRemoteDataSource().searchUsers(USER_LOGIN_1).subscribe(subscriber);
+
+        // Then
+        subscriber.awaitTerminalEvent();
+        subscriber.assertNoErrors();
+
+        List<UsersList> onNextEvents = subscriber.getOnNextEvents();
+        UsersList usersList = onNextEvents.get(0);
+        Assert.assertEquals(USER_LOGIN_1, usersList.getItems().get(0).getLogin());
+        Assert.assertEquals(USER_LOGIN_2, usersList.getItems().get(1).getLogin());
+        Mockito.verify(mNameApi).searchGithubUsers(USER_LOGIN_1);
+    }
+
+    private UsersList githubUserList() {
+        User user = new User(USER_LOGIN_1);
+
+        User user2 = new User(USER_LOGIN_2);
+
+        List<User> githubUsers = new ArrayList<>();
+        githubUsers.add(user);
+        githubUsers.add(user2);
+        UsersList usersList = new UsersList(githubUsers);
+        return usersList;
+    }
+
+    @Test
+    public void searchUsers_OtherHttpError_SearchTerminatedWithError() {
```

Comment:
ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#MethodNamingConventions